.eslintrc.js

```diff
@@ -0,0 +1,275 @@
+module.exports = {
+    'root': true,
+    'extends': [
+        'plugin:import/recommended',
+        'plugin:n/recommended',
+        'plugin:jsdoc/recommended',
+        'plugin:sonarjs/recommended'
+    ],
+    'parserOptions': {
+        'sourceType': 'module',
+        'ecmaVersion': 'latest'
+    },
+    'globals': {
+        'fetch': false,
+        'FormData': false,
+        'Headers': false,
+        'Request': false,
+        'Response': false
+    },
+    'env': {
+        'es2022': true,
+        'node': true,
+        'browser': true,
+        'worker': true,
+        'mongo': true,
+        'serviceworker': true,
+        'webextensions': true
+    },
+    'rules': {
+        'block-scoped-var': 'error',
+        'complexity': [
+            'error',
+            {
+                'max': 20
+            }
+        ],
+        'consistent-return': 'error',
+        'max-classes-per-file': [
+            'error',
+            1
+        ],
+        'max-params': [
+            'warn',
+            5
+        ],
+        'max-depth': [
+            'warn',
+            4
+        ],
+        'max-statements': [
+            'warn',
+            20
+        ],
+        'linebreak-style': [
+            'warn',
+            'unix'
+        ],
+        'class-methods-use-this': 'off',
+        'no-alert': 'error',
+        'no-caller': 'error',
+        'no-extra-label': 'error',
+        'no-labels': [
+            'error',
+            {
+                'allowLoop': false,
+                'allowSwitch': false
+            }
+        ],
+        'no-loop-func': 'error',
+        'no-prototype-builtins': 'off',
+        'no-restricted-syntax': [
+            'error',
+            'ForInStatement',
+            'LabeledStatement',
+            'WithStatement'
+        ],
+        'no-return-await': 'error',
+        'no-else-return': [
+            'error',
+            {
+                'allowElseIf': false
+            }
+        ],
+        'no-multi-assign': [
+            'error'
+        ],
+        'arrow-body-style': [
+            'error',
+            'as-needed'
+        ],
+        'no-console': 'warn',
+        'no-param-reassign': [
+            'error',
+            {
+                'props': true,
+                'ignorePropertyModificationsFor': [
+                    'acc',
+                    'accumulator',
+                    'e',
+                    'req',
+                    'request',
+                    'res',
+                    'response'
+                ]
+            }
+        ],
+        'no-useless-concat': 'error',
+        'no-nested-ternary': 'error',
+        'max-len': [
+            'error',
+            {
+                'code': 120,
+                'comments': 120,
+                'tabWidth': 4,
```

Comment:
We're using tab width 2 right?

Reply:
i will remove tabWidth, it's something this lint package shouldnt check, these formatting rules are for other linting packages.